tests/sentry/rules/conditions/test_event_attribute.py

```diff
@@ -263,34 +281,129 @@ def test_stacktrace_filename(self):
+
         rule = self.get_rule(
             data={"match": MatchType.EQUAL, "attribute": "stacktrace.module", "value": "example"}
         )
         self.assertPasses(rule, event)
 
+        rule = self.get_rule(
+            data={"match": MatchType.EQUAL, "attribute": "stacktrace.module", "value": "somecode"}
+        )
+        self.assertPasses(rule, event)
+
+        rule = self.get_rule(
+            data={"match": MatchType.EQUAL, "attribute": "stacktrace.module", "value": "othercode"}
+        )
+        self.assertPasses(rule, event)
+
         rule = self.get_rule(
             data={"match": MatchType.EQUAL, "attribute": "stacktrace.module", "value": "foo"}
         )
         self.assertDoesNotPass(rule, event)
 
     def test_stacktrace_code(self):
-        event = self.get_event()
+        """Stacktrace.code should match frames anywhere in the stack."""
+
+        event = self.get_event(
+            exception={
+                "values": [
+                    {
+                        "type": "NameError",
+                        "value": "name 'hi' is not defined",
+                        "stacktrace": {
+                            "frames": [
+                                {
+                                    "filename": "example.py",
+                                    "module": "example",
+                                    "function": "foo",
+                                    "context_line": "somecode.bar()",
+                                },
+                                {
+                                    "filename": "somecode.py",
+                                    "module": "somecode",
+                                    "function": "bar",
+                                    "context_line": "othercode.baz()",
+                                },
+                                {
+                                    "filename": "othercode.py",
+                                    "module": "othercode",
+                                    "function": "baz",
+                                    "context_line": "hi()",
+                                },
+                            ]
+                        },
+                    }
+                ]
+            }
+        )
+
         rule = self.get_rule(
             data={
                 "match": MatchType.EQUAL,
                 "attribute": "stacktrace.code",
-                "value": 'echo "hello";',
+                "value": "somecode.bar()",
             }
         )
         self.assertPasses(rule, event)
 
+        rule = self.get_rule(
+            data={
+                "match": MatchType.EQUAL,
+                "attribute": "stacktrace.code",
+                "value": "othercode.baz()",
+            }
+        )
+        self.assertPasses(rule, event)
+
+        rule = self.get_rule(
+            data={"match": MatchType.EQUAL, "attribute": "stacktrace.code", "value": "hi()"}
+        )
+        self.assertPasses(rule, event)
+
```

Comment:
Since this code is basically the same, does it make sense to have a list of `["somecode.bar()", "othercode.baz()", <etc>]` and just iterate over that? (Applies to other tests too)

Reply:
Oh, good call. Done.